### ISSUE TITLE
A single or multiple <select> with an <option> that is 'selected' and 'disabled' will not display or unselect correctly.

### ISSUE THREAD

**Issue by daschws** [wolffe/tail.select.js#1]
An example `<select>` setup below that is retrieved from a table etc. that has an `<option>` that is 'selected' and also 'disabled' should show the issue.

Let's say it's an archived invoice that we are viewing that has a part that was flagged as disabled in the system but not deleted since it is attached to other tables.

With this configuration, the initial load of the `<select>` will show 'Select an Option...' for a single and a count of '0' for a multiple `<select>` . I would instead show the text for the option 'PS3' of 'Water Inlet Valve with Mounting Bracket' for a single and a number '1' for a multiple `<select>`. Also, if you select the 'PS2' `<option>`, the 'PS3' `<option>` will also stay selected allowing for 2 selected items in a single `<select>` setup and no way to deselect it in a multiple `<select>`.

I would think that the single `<select>` should  show the selected, disabled text in the selected choice area and the multiple should show a number '1'.  Also, the disabled, selected `<option>` on a single `<select>` should only be able to be deselected if another valid `<option>` is selected, not by reselecting the disabled, selected item and on a multiple `<select>` it should be able to be deselected but not reselected since its flagged as disabled.

The code in tail.select.js in the handle function between lines 1350-1397 seems to be where the issue could be changed.

I have no fixes for this yet. Trying to find a smooth change.

Environment:
```
    Bootstrap: v5.0.0-beta3
    Chrome: Version 90.0.4430.93 (Official Build) (64-bit)
    Edge: Version 90.0.818.49 (Official build) (64-bit)
    PHP Version: 8.0.3
    Tail Version: 0.5.19
```

Sample `<select>`:
```php
<select id='testing'>
    <option value='PS1' disabled>Selector Knob with Clip</option>
    <option value='PS2'>Agitator Cam Kit</option>
    <option value='PS3' disabled selected>Water Inlet Valve with Mounting Bracket</option>
    <option value='PS4' disabled>Front Load Washer Door Lock and Switch Assembly</option>
</select>
```

Sample single javascript:
```javascript
<script>
'use strict';
document.addEventListener("DOMContentLoaded", () => {
    tail.select('#testing', {
            deselect: false, // will fail testing true or false without changes
            stayOpen: true, // set to stay open for testing only
            width: '100%', // set for testing visibilty only
        });
    });
</script>
```

Sample multiple javascript:
```javascript
<script>
'use strict';
document.addEventListener("DOMContentLoaded", () => {
    tail.select('#testing', {
            deselect: false, // will fail testing true or false without changes
            multiple: true,
            stayOpen: true, // set to stay open for testing only
            width: '100%', // set for testing visibility only
        });
    });
</script>
```

**Comment by Lenivyy** [comment]
That's an historic issue with v0.5.x (and previous) and one of many reasons why Sam and i avoided the whole option and state-handling in the 0.6 / 1.0 updates. It's hard to fix without breaking a few other things, and sadly there is no real workaround... even if you find a workaround on bottom of this.

Just to share my fuzzy memories:

The initialization of native options are passed to the [core options.set method](https://github.com/wolffe/tail.select.js/blob/main/js/tail.select.js#L338), which itself [pass them (in this case twice)](https://github.com/wolffe/tail.select.js/blob/main/js/tail.select.js#L1175) to the main state handler which [skips](https://github.com/wolffe/tail.select.js/blob/main/js/tail.select.js#L1316) disabled options and avoids a made selection. This results in an inconsistency between the native object data and the tail-stored options data, which may leads to really obscure behaviours when both are used in custom scripts.

Anyway, you cannot remove or patch the "skip" on the state-handler, because this would result into un/selecting disabled options in any way. Preventing this using the internal _init state on the select instance may would solve this issue, but may leads to other ones.

As Sam mentioned on a similar issue on the original repo:

> It was a bad design decision overtaking native behaviours. It's awful, horrible, slow, and takes dozens of lines of code for exactly zero benefits.

## Long story short, my workaround

I've used tail.select v0.5 and used a workaround similar to the following one ( sadly i don't have the original code anymore, sorry):

https://jsfiddle.net/qxhy340b/

It's ugly and may leads to other issues depending on your usage, but it may works until @wolffe comes up with v6.
